packages/beacon-node/src/chain/blocks/importBlock.ts

```diff
@@ -296,44 +292,14 @@ export async function importBlock(
   }
 
   // 6. Queue notifyForkchoiceUpdate to engine api
-  //
-  // NOTE: forkChoice.fsStore.finalizedCheckpoint MUST only change in response to an onBlock event
-  // Notifying EL of head and finalized updates as below is usually done within the 1st 4s of the slot.
-  // If there is an advanced payload generation in the next slot, we'll notify EL again 4s before next
-  // slot via PrepareNextSlotScheduler. There is no harm updating the ELs with same data, it will just ignore it.
-  if (
-    !this.opts.disableImportExecutionFcU &&
-    (newHead.blockRoot !== oldHead.blockRoot || currFinalizedEpoch !== prevFinalizedEpoch)
-  ) {
-    /**
-     * On post BELLATRIX_EPOCH but pre TTD, blocks include empty execution payload with a zero block hash.
-     * The consensus clients must not send notifyForkchoiceUpdate before TTD since the execution client will error.
-     * So we must check that:
-     * - `headBlockHash !== null` -> Pre BELLATRIX_EPOCH
-     * - `headBlockHash !== ZERO_HASH` -> Pre TTD
-     */
-    const headBlockHash = this.forkChoice.getHead().executionPayloadBlockHash ?? ZERO_HASH_HEX;
-    /**
-     * After BELLATRIX_EPOCH and TTD it's okay to send a zero hash block hash for the finalized block. This will happen if
-     * the current finalized block does not contain any execution payload at all (pre MERGE_EPOCH) or if it contains a
-     * zero block hash (pre TTD)
-     */
-    const safeBlockHash = this.forkChoice.getJustifiedBlock().executionPayloadBlockHash ?? ZERO_HASH_HEX;
-    const finalizedBlockHash = this.forkChoice.getFinalizedBlock().executionPayloadBlockHash ?? ZERO_HASH_HEX;
-    if (headBlockHash !== ZERO_HASH_HEX) {
-      this.executionEngine
-        .notifyForkchoiceUpdate(
-          this.config.getForkName(this.forkChoice.getHead().slot),
-          headBlockHash,
-          safeBlockHash,
-          finalizedBlockHash
-        )
-        .catch((e) => {
-          if (!isErrorAborted(e) && !isQueueErrorAborted(e)) {
-            this.logger.error("Error pushing notifyForkchoiceUpdate()", {headBlockHash, finalizedBlockHash}, e);
-          }
-        });
-    }
+  // This is delayed to 500ms before the next slot in PrepareNextSlotScheduler (the same to lighthouse)
+  // We should not be rush notifying EL, because if this block turns out to be a weak head and we're the next proposer,
+  // calling fcu with this block too early causes the EL to ignore the next fcu call to build block with reorg data.
+  // see https://github.com/ChainSafe/lodestar/issues/7235
+
+  // the only time we want to delay fcu is when we receive a block from gossip
+  if (!opts?.delayFcu) {
+    void this.notifyForkchoiceUpdate();
```

Comment:
If we receive a gossip block, and we are not proposer of next slot, we will never call `notifyForkchoiceUpdate`. Is this correct?

Reply:
> we will never call notifyForkchoiceUpdate. Is this correct?

we call it at 11.5 seconds into the slot

Reply:
> > we will never call notifyForkchoiceUpdate. Is this correct?
> 
> we call it at 11.5 seconds into the slot

Yea only when we are proposing next slot. 

My question was if we are not proposing next slot

Reply:
I misread. Ignore this